spec/unit/lib/temporal/testing/local_workflow_context.rb

```diff
@@ -0,0 +1,97 @@
+require 'temporal/testing'
+require 'temporal/workflow'
+require 'temporal/api/errordetails/v1/message_pb'
+
+describe Temporal::Testing::LocalWorkflowContext do
+  let(:namespace) { 'default-namespace' }
+  let(:task_queue) { 'default-task-queue' }
+
+  class TestFailedActivity < Temporal::Activity
+    namespace namespace
+    task_queue task_queue
+
+    def execute
+      raise 'oops'
+    end
+  end
+
+  class TestActivity < Temporal::Activity
+    namespace namespace
+    task_queue task_queue
+
+    def execute
+      'ok'
+    end
+  end
+
+  class TestAsyncActivity < Temporal::Activity
+    namespace namespace
+    task_queue task_queue
+
+    def execute
+      activity.async
+    end
+  end
+
+  describe 'execute_activity' do
+    let(:workflow_id) { 'workflow_id_1' }
+    let(:run_id) { 'run_id_1' }
+    let(:execution) { Temporal::Testing::WorkflowExecution.new }
+    let(:workflow_context) do
+      Temporal::Testing::LocalWorkflowContext.new(
+        execution,
+        workflow_id,
+        run_id,
+        []
+      )
+    end
+    let(:async_token) do
+      Temporal::Activity::AsyncToken.encode(
+        namespace,
+        1, # activity ID starts at 1 for each workflow
+        workflow_id,
+        run_id
+      )
+    end
+    describe 'outcome is captured in the future' do
```

Comment:
Should we also have a spec for a failure in a typical, non-futures case, to make sure the exception gets re-raised?

Reply:
Sure, I'll add that too. That should cover the case in unit tests that's currently only being covered by the examples integration tests.